GOVERNANCE_DOC.md

```diff
@@ -183,9 +177,83 @@ Additionaly, four lists will be linked to each proposal:
 - `InitVotingPowerList`: Snapshot of validators' voting power **when proposal enters voting period** (only saves validators whose voting power is >0).
 - `MinusesList`: List of minuses for each validator. Used to compute validators' voting power when they cast a vote.
 
+*Note: Actual data structure will probably not be list. These lists are here to illustrate what needs to be stored, not how*
+
 Two final parameters, `InitTotalVotingPower` and `InitProcedureNumber` associated with `proposalID` will be saved when proposal enters voting period.
 
 We also introduce `ProposalProcessingQueue` which lists all the `ProposalIDs` of proposals that reached `MinDeposit` from oldest to newest. Each round, the oldest element of `ProposalProcessingQueue` is checked during `BeginBlock` to see if `CurrentBlock == VotingStartBlock + InitProcedure.VotingPeriod`. If it is, then the application checks if validators in `InitVotingPowerList` have voted and, if not, applies `GovernancePenalty`. After that proposal is ejected from `ProposalProcessingQueue` and the new first element of the queue is evaluated. Note that if a proposal is urgent and accepted under the special condition, its `ProposalID` must be ejected from `ProposalProcessingQueue`.
+The beginning and the end of the `ProposalProcessingQueue` are identified using `ProposalProcessingQueueBeginning` and `ProposalProcessingQueueEnd`.
+
+And the pseudocode for the `ProposalProcessingQueue`:
+
+```
+  in BeginBlock do 
+    
+    checkProposal()  
+    
+    
+    
+  func checkProposal()  
+    if (ProposalProcessingQueueBeginning == ProposalProcessingQueueEnd)
+      return
+
+    else
+      retrieve proposalID from ProposalProcessingQueue[ProposalProcessingQueueBeginning]
+      retrieve proposal from proposalID
+      retrieve initProcedureNumber from proposalID
+      retrieve initProcedure from initProcedureNumber
+
+      if (CurrentBlock == proposal.VotingStartBlock + initProcedure.VotingPeriod)
+        retrieve initVotingPowerList from proposalID
+        retrieve votersList from proposalID
+        retrieve validators from initVotingPowerList
+
+        for each validator in validators
+          if validator is not in votersList
+            slash validator by ActiveProcedure.GovernancePenalty
```

Comment:
Can we accomplish this without a for loop? I guess it doesn't really matter here

Reply:
don't think so

Reply:
I mean I'm sure there is a way we could accomplish that if we needed to - I'm just not convinced we really care because, unlike block processing, this process doesn't happen very often. We're good..